src/optimum/rbln/transformers/models/decoderonly/decoderonly_architecture.py

```diff
@@ -222,16 +224,82 @@ def phase(self, phase: str):
 
-    def forward_common(
-        self,
-        input_ids_or_inputs_embeds: torch.Tensor,
-        cache_position: torch.Tensor,
-        attention_mask: torch.Tensor,
-        query_position: torch.Tensor,
-        block_tables: torch.Tensor,
-        rotary_emb: Union[nn.Module, torch.Tensor],
-        *past_key_values: List[torch.Tensor],
-    ):
+    def convert_args(self, *args):
+        """Converts variable input arguments into a standardized tuple for the forward pass.
+
+        Args:
+            *args: Variable arguments in the following order:
+                - input_ids_or_inputs_embeds
+                - cache_position
+                - block_tables
+                - (query_position, if phase == "prefill")
+                - (attention_mask, if use_attention_mask is True)
+                - (position_ids, if use_position_ids is True)
+                - *past_key_values (remaining arguments)
+
+        Returns:
+            tuple: (input_ids_or_inputs_embeds, cache_position, block_tables, query_position,
+                    attention_mask, position_ids, past_key_values, rotary_emb)
+
+        Raises:
+            ValueError: If phase is invalid or required arguments are missing.
+        """
+        if self.phase not in ["decode", "prefill"]:
+            raise ValueError(f"Unknown phase: {self.phase}")
+
+        (input_ids_or_inputs_embeds, cache_position, block_tables, *conditional_args) = args
+
+        query_position = None
+        attention_mask = None
+        position_ids = None
+        arg_idx = 0
+        if self.phase == "prefill":
+            if arg_idx >= len(conditional_args) - (2 * self.num_hidden_layers):
+                raise ValueError("Missing query_position for prefill phase")
+            query_position = conditional_args[arg_idx]
+            arg_idx += 1
+
+        if self.use_attention_mask:
+            if arg_idx >= len(conditional_args) - (2 * self.num_hidden_layers):
+                raise ValueError("Missing attention_mask when use_attention_mask is True")
+            attention_mask = conditional_args[arg_idx]
+            arg_idx += 1
+
+        if self.use_position_ids:
+            if arg_idx >= len(conditional_args) - (2 * self.num_hidden_layers):
+                raise ValueError("Missing position_ids when use_position_ids is True")
+            position_ids = conditional_args[arg_idx]
+            arg_idx += 1
+
+        past_key_values = conditional_args[arg_idx:]
+        if len(past_key_values) != 2 * self.num_hidden_layers:
+            raise ValueError(
+                f"Different past_key_values to model's config. {len(past_key_values)} != {2 * self.num_hidden_layers}"
+            )
+
+        return (
+            input_ids_or_inputs_embeds,
+            cache_position,
+            block_tables,
+            query_position,
+            attention_mask,
+            position_ids,
+            past_key_values,
+            self.rotary_emb,
+        )
```

Comment:
how about like this?


```suggestion
    def convert_args(self, *args: List[Any]):
        input_ids = None if self.use_inputs_embeds else args.pop(0)
        inputs_embeds = args.pop(0) if self.use_inputs_embeds else None
        cache_position = args.pop(0)
        block_tables = args.pop(0)
        query_position = args.pop(0) if self.phase == "prefill" else None
        attention_mask = args.pop(0) if self.use_attention_mask else None
        position_ids = args.pop(0) if self.use_position_ids else None
        past_key_values = args

        return (
            input_ids,
            inputs_embeds,
            cache_position,
            block_tables,
            query_position,
            attention_mask,
            position_ids,
            past_key_values,
        )
```